src/components/Form/Form.tsx

```diff
@@ -14,7 +14,11 @@ type Props<T extends FormDetails> = {
   defaults: T;
   asyncGetDefaults?: (() => Promise<T>) | false;
   validate?: (form: T) => FormErrors<T>;
-  onSubmit: (form: T) => Promise<FormErrors<T> | void>;
+  onSubmit: (
+    form: T,
+    source?: string,
+    discharged?: boolean,
```

Comment:
passing "discharged" in a generic reusable component is not ideal.

```suggestion
```

Reply:
without It I cannot use discharged in [ShiftingDetailsUpdate.tsx](https://github.com/kihan2518B/care_fe/blob/develop/src/components/Shifting/ShiftDetailsUpdate.tsx) component.

Reply:
i don't see any usages that is passing the third argument to this function when invoking. Can you point it out if present?

Reply:
I will remove it